k8s/src/main/scala/io/buoyant/k8s/resources.scala

```diff
@@ -113,51 +130,49 @@ class NsThirdPartyVersion[O <: KubeObject](client: Client, owner: String, ownerV
  * `/api/v1/namespaces/{namespace}/endpoints`.
  */
 private[k8s] class ListResource[O <: KubeObject: TypeReference, W <: Watch[O]: TypeReference, L <: KubeList[O]: TypeReference](
-  val client: Client,
-  basePath: String,
+  parent: Resource,
   protected val backoffs: Stream[Duration] = Watchable.DefaultBackoff,
   protected val stats: StatsReceiver = DefaultStatsReceiver
-)(implicit od: ObjectDescriptor[O, W]) extends Watchable[O, W] with Resource {
-  val name = implicitly[ObjectDescriptor[O, W]].listName
-  val path = s"$basePath/$name"
-
-  def get(
-    labelSelector: Option[String] = None,
-    fieldSelector: Option[String] = None,
-    resourceVersion: Option[String] = None,
-    retryIndefinitely: Boolean = false
-  ): Future[L] = {
-    val req = Api.mkreq(http.Method.Get, this.path, None,
-      "labelSelector" -> labelSelector,
-      "fieldSelector" -> fieldSelector,
-      "resourceVersion" -> resourceVersion)
-    val retry = if (retryIndefinitely) infiniteRetryFilter else Filter.identity[http.Request, http.Response]
-    val retryingClient = retry andThen client
-    Trace.letClear(retryingClient(req)).flatMap(Api.parse[L])
-  }
-
-  protected def restartWatches(
-    labelSelector: Option[String] = None,
-    fieldSelector: Option[String] = None
+)(implicit od: ObjectDescriptor[O, W])
+  extends Watchable[O, W, L]
+  with Resource {
+
+  override val client: Client = parent.client
+  val name: String = implicitly[ObjectDescriptor[O, W]].listName
+  final override val path = s"${parent.path}/$name"
+  final override val watchPath = s"${parent.watchPath}/$name"
+  /**
+   * @return a Future containing a sequence of Watches and an optional String representing the current resourceVersion
+   */
+  override protected def restartWatches(
+    labelSelector: Option[String],
+    fieldSelector: Option[String]
   ): Future[(Seq[W], Option[String])] =
-    get(labelSelector, fieldSelector, None, true).map { list =>
-      (list.items.map(od.toWatch), list.metadata.flatMap(_.resourceVersion))
-    }
+    get(
+      labelSelector,
+      fieldSelector,
+      None,
+      retryIndefinitely = true,
+      watch = true
+    )
+      .map { list =>
```

Comment:
IntelliJ's formatting tends to do this, but I think scalariform should be ok with having them on the same line.